test/cases/schema-title-as-type-name.ts

```diff
@@ -0,0 +1,48 @@
+export var schema = {
+    "$schema": "http://json-schema.org/draft-03/schema",
+    "id": "http://mycompany.com/api/referencing.json",
+    "title": "Referencing",
+    "type": "object",
+    "properties": {
+        "ref": {
+            "$ref": "../../test/resources/ReferencedType.json"
```

Comment:
Again, no need for the $ref, I think. Since you're testing that `title` takes precedence over `id`, your test case should be as slim as possible and just test that one behavior.

Concretely, if you replace the `"ref"` property with a simple `"foo": "number"` or something like that, it would simplify your test and focus on the behavior that you're actually testing.